### PR TITLE
[#491] Error snackbar when stream statistics can't be fetched 

### LINKED ISSUE #491: Error snackbar displayed when stream statistics can't be fetched on Stream Details Page

### PR DESCRIPTION
**Pull request by sharon-ho**
# General info

**Issue number**: Closes #491

**Task description**: 
- Add error snackbar displayed on Stream Details Page when stream statistics can't be fetched

# Testing

**Test coverage**: 100%

# Additional notes

**Note to reviewers**:
N/A

**To do before merging**:
- Merge #490 first since stream model is updated in that PR

### CODE REVIEW

**Review by gdzooks — DISMISSED** [review]
simple & clean! :3

**Review by gdzooks — APPROVED** [review]

File: frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx [hunk 1]
```diff
@@ -1,131 +1,165 @@
...
+          "handleStatsChange"
+        );
+
+        wrapper.instance().componentDidMount();
+
+        expect(StreamApi.getStreamStatistics).toHaveBeenCalledWith(
+          dummyStream.id
+        );
+
+        await new Promise(setImmediate);
```
Comment (FGRCL) [inline frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx:73]: What does that do? create a promise then wait for it to be resolved instantly?

File: frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx [hunk 2]
```diff
@@ -1,131 +1,165 @@
...
-      const expectedProps = { item: true, xs: 12 };
-      const gridProps = wrapper.find(Grid).at(0).props();
-
-      expect(gridProps.item).toBe(expectedProps.item);
-      expect(gridProps.xs).toBe(expectedProps.xs);
+    describe("getButton() function", () => {
+      it("it should return appropriate button", () => {
+        const actualButton = wrapper.instance().getButton();
+        expect(actualButton).toStrictEqual(expectedButton);
+      });
```
Comment (FGRCL) [inline frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx:132]: Did  you remove the test when these two methods return null?

File: frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx (hunk 1 above)
Comment (felixlapierre) [reply frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx:73]: waits for all promises to resolve, necessary for the code under test to finish before we assert. (discussed in person)

File: frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx (hunk 2 above)
Comment (FGRCL) [reply frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx:132]: I talked to @sharon-ho and apprently the removed tests test for stuff that will never happen so it's all good!

**Review by FGRCL — APPROVED** [review]
Sweet addition!

**Review by felixlapierre — APPROVED** [review]

**Pull request merged**